api/services/gameService.js

```diff
@@ -34,43 +34,33 @@ async function fetchSpectatorUsernames(gameId) {
 
 // Used to create fully populated game
 function tempUser(usr, points) {
-  this.pNum = usr.pNum;
-  this.hand = usr.hand;
-  this.points = points;
-  this.faceCards = usr.faceCards;
-  this.frozenId = usr.frozenId;
-  this.id = usr.id;
-  this.username = usr.username;
-
-  this.hand.sort(comapreByRankThenSuit);
-  this.points.sort(comapreByRankThenSuit);
-  this.faceCards.sort(comapreByRankThenSuit);
+  Object.entries(usr).forEach(([key,val]) => {
+    if(key === 'encryptedPassword'){
+      return;
+    }
+    if(key === 'points'){
+      this[key] = points;
+      return;
+    }
+    this[key] = val;
+  });
+  this.hand.sort(compareByRankThenSuit);
+  this.points.sort(compareByRankThenSuit);
+  this.faceCards.sort(compareByRankThenSuit);
 }
 
-function tempGame(game, spectators, p0, p1) {
-  this.id = game.id;
-  this.players = [p0, p1];
-  this.spectatingUsers = spectators;
-  this.deck = game.deck;
-  this.scrap = game.scrap;
-  this.topCard = game.topCard;
-  this.secondCard = game.secondCard;
-  this.oneOff = game.oneOff;
-  this.oneOffTarget = game.oneOffTarget;
-  this.twos = game.twos;
-  this.log = game.log;
-  this.chat = game.chat;
-  this.id = game.id;
-  this.turn = game.turn;
-  this.passes = game.passes;
-  this.resolving = game.resolving;
-  this.lastEvent = game.lastEvent;
-  this.result = game.result;
-  this.isRanked = game.isRanked;
-  this.p0 = game.p0;
-  this.p1 = game.p1;
-  this.p0Ready = game.p0Ready;
-  this.p1Ready = game.p1Ready;
+function tempGame(game, spectatingUsers, p0, p1) {
+  Object.entries(game).forEach(([key, val]) => {
+    if(key === 'players'){
+      this[key] = [p0, p1];
+      return;
+    }
+    if(key === 'spectatingUsers'){
+      this[key] = spectatingUsers;
+      return;
+    }
+    this[key] = val;
```

Comment:
I know this is the approach prescribed by the issue, but upon reflection I think it might actually be clearer to use destructuring and spread syntax to achieve omitting/manipulating certain specific properties. 

First to get on the same page about the purpose of this function i.e. why not just use the game object itself without this transform, it's to replace the `players` attribute because by default when sails queries for the game and populates it's players, the players themselves don't get populated with their hands, faceCards, etc. So this function combines the main `game` object with the deeply populated versions of the players (p1 and p0). With that in mind, I think the *only* change we need to make to the game is to reassign its players, so we can probably simplify this. As I write it, I can think of a couple approaches we can take, and I'm not actually sure what is best. Let's discuss them together. @seriouslysean penny for your thoughts as well.

Reply:
The simplest approach is to mutate the `game` argument in-place and return it. 

```
function tempGame(game, spectatingUsers, p0, p1) {
  game.players = [p0, p1];
  game.spectatingUsers = spectatingUsers;
  return game;
}

```

This has the advantage of being dead-simple, and the disadvantage of mutating an argument, which in principle can have unintended side effects. Specifically it doesn't just return the desired data, it updates the variable `game` to match the same thing as well. This is probably fine for Cuttle because it's not necessarily bad that `game` become the fully populated game, but generally it's a no-no because you can easily get confused why variables are changing

Reply:
What might be better would be to return a new object that copies the game properties with spread syntax and overrides the specific new ones that we want to change:

```
function tempGame(game, spectatingUsers, p0, p1) {
  return {
    ...game,
    players: [p0, p1],
    spectatingUsers,
  };
}
```

This is probably better because `game` remains unchanged and we just get a new object returned without cross-contamination. It's 2 more lines and requires understanding spread syntax + the fact that the properties specified on the bottom take precedence, but it's still succinctly clear

Reply:
Alternatively we could deep copy the game and then mutate properties on the copy:

```

function tempGame(game, spectatingUsers, p0, p1) {
  const res = _.cloneDeep(game);
  res.players = [p0, p1];
  res.spectatingUsers = spectatingUsers;
  return game;
}

```
This fixes the argument mutation but I think the 2nd way is cleaner , as this requires a lodash method

Reply:
I think the second or the third one are good, second is cleaner, third is slightly more obvious what is happening. On the third one I think we could use structuredClone() instead of cloneDeep this way you don't need lodash for it.

Reply:
Yes let's go with approach 2

Reply:
Honestly in this case, do we even need this function? I think we could just return it at the end of populateGame().